### ISSUE TITLE
Error casting types 

### ISSUE THREAD

**Issue by gh-pp** [JetBrains/YouTrackSharp#112]
## Expected behavior
Valid casting EnumBundleElement

## Actual behavior
While Im trying to fetch data via issueService.GetIssue  after upgrade to version 2021.3.3 I get an error
```
Unable to cast object of type 'YouTrackSharp.Generated.EnumBundleElement' to type 'YouTrackSharp.Generated.StateBundleElement
```

**Comment by rekolobov** [comment]
@gh-pp Could you please provide a bit more details about configuration and values of particular field in your tracker? Thanks!

**Comment by gh-pp** [comment]
Thats is what i've found in api response:
<img width="385" alt="Screenshot 2021-07-29 at 11 10 37" src="https://user-images.githubusercontent.com/88095885/127465153-c64194d5-97a8-455c-85df-0970c6e81bb4.png">

**Comment by rekolobov** [comment]
Released [2021.3.4](https://github.com/JetBrains/YouTrackSharp/releases/tag/2021.3.4) and published to NuGet

Please check if your issue was indeed diagnosed correctly and fixed

**Comment by gh-pp** [comment]
Everything works fine, thank you

**Comment by robbrunato** [comment]
We are seeing the opposite trying to get an issue with a state machine attached. 
![image](https://user-images.githubusercontent.com/86723334/208502844-ec5ec34d-3b19-4500-8a56-1d498a53681e.png)

If I remove the state machine we are able to get the issue. YouTrackSharp 2022.3.0 and YouTrack 2022.3.63553.

**Comment by rekolobov** [comment]
@robbrunato Hello, thank you for the report and sorry for the extremely late reply.
I've committed the fix in ae84138575a0bcd08414805a87bf626367a87036 along with the tests for all field types to not miss similar stuff again. Expecting to release it in the scope of 2022.3.1 tomorrow
